index.js

```diff
@@ -188,7 +190,7 @@ Compiler.prototype.compileMessage = function(message, root) {
   return res;
 };
 
-module.exports = function(filename, model) {
+module.exports = function(filename, model,allow_additional_props) {
   var compiler = new Compiler(filename);
-  return compiler.compile(model);
+  return compiler.compile(model,allow_additional_props);
```

Comment:
Yeah, this is a problem that can also happen in another method.. I think it's resolve. How about we make the `opts` solution you described above the 1st argument in all these methods. That way you have an easy way to support other flags later on. Opinion ?